src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,231 @@
+import { toast } from 'common/helpers/toast/toast';
+import { AxiosError } from 'axios';
+import { useTransactionQuery } from '../common/queries';
+import { DecimalNumberInput } from 'components/forms/DecimalNumberInput';
+import { DecimalInputSeparators } from 'common/interfaces/decimal-number-input-separators';
+import { useResolveCurrency } from 'common/hooks/useResolveCurrency';
+
+export function Edit() {
+  const { t } = useTranslation();
+
+  const navigate = useNavigate();
+
+  const { id } = useParams<string>();
+
+  const currencies = useCurrencies();
+
+  const resolveCurrency = useResolveCurrency();
+
+  const { documentTitle } = useTitle('edit_transaction');
+
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const [errors, setErrors] = useState<TransactionValidation>();
+
+  const { data: response } = useTransactionQuery({ id });
+
+  const [transaction, setTransaction] = useState<TransactionInput>();
+
+  const [currencySeparators, setCurrencySeparators] = useState<
+    DecimalInputSeparators | undefined
+  >();
+
+  const pages = [
+    { name: t('transactions'), href: '/transactions' },
+    { name: t('edit_transaction'), href: `/transactions/${id}/edit` },
+  ];
+
+  const getCurrencySeparators = (currencyId: string) => {
+    const currency = resolveCurrency(currencyId) || currencies[0];
+    return {
+      decimalSeparator: currency?.decimal_separator,
+      precision: currency?.precision,
+      thousandSeparator: currency?.thousand_separator,
+    };
+  };
+
+  const getTransactionInputObject = (
+    responseDetails: TransactionResponse | undefined
+  ) => {
+    const { date, amount, currency_id, bank_integration_id, description } =
+      responseDetails || {};
+    setCurrencySeparators(getCurrencySeparators(currency_id || ''));
+    return {
+      base_type:
+        responseDetails?.base_type === 'CREDIT' ? 'deposit' : 'withdrawal',
+      date,
+      amount,
+      currency_id,
+      bank_integration_id,
+      description,
+    };
+  };
+
+  const handleChange = (
+    property: keyof TransactionInput,
+    value: TransactionInput[keyof TransactionInput]
+  ) => {
+    if (property === 'currency_id') {
+      setCurrencySeparators(getCurrencySeparators(value?.toString() || ''));
+    }
+
+    setTransaction(
+      (prevState) => prevState && { ...prevState, [property]: value }
+    );
+  };
+
+  const onSave = async (event: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+
+    setErrors(undefined);
+
+    setIsSaving(true);
+
+    const toastId = toast.processing();
+
+    try {
+      await request('PUT', endpoint('/api/v1/bank_transactions/:id', { id }), {
+        ...transaction,
+        amount: Number(transaction?.amount),
+        base_type: transaction?.base_type === 'deposit' ? 'CREDIT' : 'DEBIT',
+      });
+
+      toast.success(t('updated_transaction'), { id: toastId });
+      setIsSaving(false);
+      navigate(route('/transactions'));
+    } catch (cachedError) {
```

Comment:
Please rename this variable. Not sure what we are caching here.

Reply:
> Please rename this variable. Not sure what we are caching here.

renamed